api/utilities/response_generation.py

```
@@ -0,0 +1,445 @@
+            result["quality_score"] -= 0.3
+            result["feedback"].append("Low overlap with knowledge base")
+
+    # 3. Check response length (too short responses are often low quality)
+    if len(response.split()) < 7:
+        result["is_unsatisfactory"] = True
+        result["quality_score"] -= 0.2
+        result["feedback"].append("Response is too short")
+
+    # 4. Check if response addresses user question (very basic check)
+    user_words = set(user_message.lower().split())
+    important_user_words = [w for w in user_words if len(w) > 3 and w not in ["what", "when", "where", "which", "how", "would", "could", "should", "about", "with", "this", "that", "there", "their", "have", "your"]]
```

Ruff (0.11.9)

Local variable `important_user_words` is assigned to but never used

Remove assignment to unused variable `important_user_words`

(F841)